crate-tmp/src/nvtxw.rs

```diff
@@ -0,0 +1,280 @@
+pub use crate::nvtxw_bindings::nvtxwInterfaceCore_t as InterfaceHandle;
+pub use crate::nvtxw_bindings::nvtxwResultCode_t as ResultCode;
+use crate::nvtxw_bindings::nvtxwSessionAttributes_t as SessionAttributes;
+pub use crate::nvtxw_bindings::nvtxwSessionHandle_t as SessionHandle;
+use crate::nvtxw_bindings::nvtxwStreamAttributes_t as StreamAttributes;
+pub use crate::nvtxw_bindings::nvtxwStreamHandle_t as StreamHandle;
+
+use crate::nvtxw_bindings::NVTXW3_INIT_MODE_LIBRARY_FILENAME;
+use crate::nvtxw_bindings::NVTXW3_INTERFACE_ID_CORE_V1;
+use crate::nvtxw_bindings::NVTXW3_RESULT_FAILED;
+use crate::nvtxw_bindings::NVTXW3_RESULT_INVALID_ARGUMENT;
+use crate::nvtxw_bindings::NVTXW3_RESULT_SUCCESS;
+use crate::nvtxw_bindings::NVTXW3_STREAM_ORDERING_SKID_NONE;
+
+pub const NVTX_PAYLOAD_ENTRY_FLAG_POINTER: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_FLAG_POINTER as u64;
+pub const NVTX_PAYLOAD_ENTRY_FLAG_EVENT_MESSAGE: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_FLAG_EVENT_MESSAGE as u64;
+pub const NVTX_PAYLOAD_ENTRY_FLAG_EVENT_TIMESTAMP: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_FLAG_EVENT_TIMESTAMP as u64;
+pub const NVTX_PAYLOAD_ENTRY_FLAG_RANGE_BEGIN: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_FLAG_RANGE_BEGIN as u64;
+pub const NVTX_PAYLOAD_ENTRY_FLAG_RANGE_END: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_FLAG_RANGE_END as u64;
+pub const NVTX_PAYLOAD_ENTRY_TYPE_UINT64: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_TYPE_UINT64 as u64;
+pub const NVTX_PAYLOAD_ENTRY_TYPE_NVTX_COLOR: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_TYPE_NVTX_COLOR as u64;
+pub const NVTX_PAYLOAD_ENTRY_TYPE_CSTRING: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_ENTRY_TYPE_CSTRING as u64;
+pub const NVTX_PAYLOAD_SCHEMA_TYPE_STATIC: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_TYPE_STATIC as u64;
+pub const NVTX_PAYLOAD_SCHEMA_FLAG_NONE: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_FLAG_NONE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_NAME: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_NAME as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_TYPE: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_TYPE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_FLAGS: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_FLAGS as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_ENTRIES: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_ENTRIES as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_NUM_ENTRIES: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_NUM_ENTRIES as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_STATIC_SIZE: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_STATIC_SIZE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_SCHEMA_ID: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_SCHEMA_ID as u64;
+
+pub const NVTXW3_STREAM_ORDER_INTERLEAVING_NONE: i16 = 0;
+pub const NVTXW3_STREAM_ORDERING_TYPE_UNKNOWN: i16 = 0;
+
+pub fn initialize_simple() -> Result<InterfaceHandle, ResultCode> {
+    let so_name = "libNvtxwBackend.so";
```

Comment:
Is this going to load an `so` dynamically? Is that it?

Reply:
Yes, that is correct.